jobs/src/main/scala/dcos/metronome/repository/impl/kv/marshaller/JobSpecMarshaller.scala

```diff
@@ -136,6 +136,7 @@ object RunSpecConversions {
       runSpec.args.foreach { args => builder.addAllArguments(args.asJava) }
       runSpec.user.foreach(builder.setUser)
       runSpec.docker.foreach { docker => builder.setDocker(docker.toProto) }
+      runSpec.taskKillGracePeriodSeconds.foreach { killGracePeriod => builder.setTaskKillGracePeriodSeconds(killGracePeriod.toMillis) }
```

Comment:
Wait a _second_ – shouldn't this be **`toSeconds`**? 🙄 

Also, you're missing the conversion in `implicit class ProtoToRunSpec` on line 145ff.

(In the current impl the gracePeriod would be received via the API, stored in zk and passed over to Marathon, but not retrieved from zk after a failover).